### ISSUE TITLE
Payload and Evasion modules don't validate the datastore with to_handler

### ISSUE THREAD

**Issue by zeroSteiner** [rapid7/metasploit-framework#17935]
The `to_handler` command available in payload modules does not validate the datastore options prior to starting the handler. This means if an option is missing, the user will still see a message indicating that the handler started successfully.

In the following scenario, the `LHOST` option is required. The `generate` command fails because `LHOST` failed to validate, but `to_hander` does not indicate any error.

```
msf6 payload(linux/x64/meterpreter/reverse_tcp) > show options 

Module options (payload/linux/x64/meterpreter/reverse_tcp):

   Name   Current Setting  Required  Description
   ----   ---------------  --------  -----------
   LHOST                   yes       The listen address (an interface may be specified)
   LPORT  4444             yes       The listen port


View the full module info with the info, or info -d command.

msf6 payload(linux/x64/meterpreter/reverse_tcp) > to_handler 
[*] Payload Handler Started as Job 
msf6 payload(linux/x64/meterpreter/reverse_tcp) > generate -f raw
[-] Payload generation failed: One or more options failed to validate: LHOST.
msf6 payload(linux/x64/meterpreter/reverse_tcp) > 
```

Evasion modules seem to suffer from the same issue. In this case, I'd again expect a failure because the required `LHOST` option is blank.

```
use evasion/windows/syscall_inject 
[*] Using configured payload windows/x64/meterpreter/reverse_tcp
msf6 evasion(windows/syscall_inject) > show options 

Module options (evasion/windows/syscall_inject):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   CIPHER    chacha           yes       Shellcode encryption type (Accepted: chacha, rc4)
   FILENAME  Xtr.exe          yes       Filename for the evasive file (default: random)
   SLEEP     20000            no        Sleep time in milliseconds before executing shellcode


Payload options (windows/x64/meterpreter/reverse_tcp):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   EXITFUNC  process          yes       Exit technique (Accepted: '', seh, thread, process, none)
   LHOST                      yes       The listen address (an interface may be specified)
   LPORT     4444             yes       The listen port


Evasion target:

   Id  Name
   --  ----
   0   Microsoft Windows (x64)



View the full module info with the info, or info -d command.

msf6 evasion(windows/syscall_inject) > to_handler 
[*] Payload Handler Started as Job 
msf6 evasion(windows/syscall_inject) > 
```

**Comment by saurabhmj11** [comment]
It seems that the to_handler command in both payload and evasion modules does not validate the datastore options prior to starting the handler, which could lead to issues when required options are not set. In your first example, the generate command failed because the LHOST option was missing, but to_handler did not indicate any error. Similarly, in the second example, the required LHOST option was blank, but to_handler still started the handler successfully.

To avoid such issues, it's important to ensure that all required options are set correctly before starting the handler. You can use the show options command to view the required options for a specific module, and then use the set command to set their values. Once all required options are set, you can use the to_handler command to start the handler.

In summary, it's always a good practice to double-check that all required options are set correctly before starting a handler to avoid any unexpected issues.
